diff --git a/dist/index.cjs b/dist/index.cjs
index eb088ed49d73d681f4e9a8ae4f7c34998a65a089..ad69b67cdbc907df5c4aa6360b8ec30569b23cdc 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -599,6 +599,9 @@ class OpenAPIRegistry {
         };
     }
     schemaWithRefId(refId, zodSchema) {
+        if (!("openapi" in zodSchema)) {
+            return zodSchema.meta({ id: refId });
+        }
         return zodSchema.openapi(refId);
     }
 }
diff --git a/dist/index.mjs b/dist/index.mjs
index 57373e21df12b7a399e4af84301483ae314d62c0..cb1d552897af478575872cc7deeb3bf46afc080c 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -597,6 +597,9 @@ class OpenAPIRegistry {
         };
     }
     schemaWithRefId(refId, zodSchema) {
+        if (!("openapi" in zodSchema)) {
+            return zodSchema.meta({ id: refId });
+        }
         return zodSchema.openapi(refId);
     }
 }
