diff --git a/dist/index.cjs b/dist/index.cjs
index c8778b90d64fc659d707436b1b51bbbbf5b5feab..96bfc548123c8473a731640a9f98c58928a69330 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -599,6 +599,9 @@ class OpenAPIRegistry {
         };
     }
     schemaWithRefId(refId, zodSchema) {
+        if (!("openapi" in zodSchema)) {
+            return zodSchema.meta({ id: refId });
+        }
         return zodSchema.openapi(refId);
     }
 }
diff --git a/dist/index.mjs b/dist/index.mjs
index b3b7b28db3e5c6b21aced00b44f972726f7d627d..097f92d06c3c44e89bce3ea7f3d250b2c3de531c 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -597,6 +597,9 @@ class OpenAPIRegistry {
         };
     }
     schemaWithRefId(refId, zodSchema) {
+        if (!("openapi" in zodSchema)) {
+            return zodSchema.meta({ id: refId });
+        }
         return zodSchema.openapi(refId);
     }
 }
