diff --git a/dist/index.cjs b/dist/index.cjs
index 1123df3551fa7b0cf7ddf0fee7f4cc080d2ed18d..f7806dfa36376c3766e060de1746614ca2809605 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -599,6 +599,9 @@ class OpenAPIRegistry {
         };
     }
     schemaWithRefId(refId, zodSchema) {
+        if (!("openapi" in zodSchema)) {
+            return zodSchema.meta({ id: refId });
+        }
         return zodSchema.openapi(refId);
     }
 }
diff --git a/dist/index.mjs b/dist/index.mjs
index 59baab9d550cd4d5ac28b84156c4d86959db3165..9e45a0cbf5fcbf6cdcbe39c29b7b521df327f302 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -597,6 +597,9 @@ class OpenAPIRegistry {
         };
     }
     schemaWithRefId(refId, zodSchema) {
+        if (!("openapi" in zodSchema)) {
+            return zodSchema.meta({ id: refId });
+        }
         return zodSchema.openapi(refId);
     }
 }
