diff --git a/dist/changesets-get-github-info.cjs.js b/dist/changesets-get-github-info.cjs.js
index a74df59f8a5988f458a3476087399f5e6dfe4818..486d9c2f5926f64584a0df8b3fed1b223bd1d73c 100644
--- a/dist/changesets-get-github-info.cjs.js
+++ b/dist/changesets-get-github-info.cjs.js
@@ -167,7 +167,7 @@ const GHDataLoader = new DataLoader__default["default"](async requests => {
 
     repos[repo].push(data);
   });
-  const data = await fetch__default["default"]("https://api.github.com/graphql", {
+  let data = await fetch__default["default"]("https://api.github.com/graphql", {
     method: "POST",
     headers: {
       Authorization: `Token ${process.env.GITHUB_TOKEN}`
@@ -175,7 +175,11 @@ const GHDataLoader = new DataLoader__default["default"](async requests => {
     body: JSON.stringify({
       query: makeQuery(repos)
     })
-  }).then(x => x.json());
+  }).then(x => x.text());
+
+  console.log(data)
+
+  data = JSON.parse(data);
 
   if (data.errors) {
     throw new Error(`An error occurred when fetching data from GitHub\n${JSON.stringify(data.errors, null, 2)}`);
